cv_bridge/include/cv_bridge/cv_bridge.h

```diff
@@ -136,11 +150,25 @@ class CV_BRIDGE_EXPORT CvImage
    * can be: jpg, jp2, bmp, png, tif at the moment
    * support this format from opencv:
    * http://docs.opencv.org/modules/highgui/doc/reading_and_writing_images_and_video.html#Mat imread(const string& filename, int flags)
+   * params are the options for cv::imencode.
+   * Default value is empty list.
    */
   void toCompressedImageMsg(
     sensor_msgs::msg::CompressedImage & ros_image,
-    const Format dst_format = JPG) const;
+    const Format dst_format = JPG, const std::vector<int> &params=std::vector<int>()) const;
 
+  /**
+   * dst_format is compress the image to desire format.
+   * Default value is empty string that will convert to jpg format.
+   * can be: jpg, jp2, bmp, png, tif at the moment
+   * support this format from opencv:
+   * http://docs.opencv.org/modules/highgui/doc/reading_and_writing_images_and_video.html#Mat imread(const string& filename, int flags)
+   * params are the options for cv::imencode.
+   * Default value is empty list.
+   */
+  void toCompressedImageMsg(
+    sensor_msgs::msg::CompressedImage & ros_image,
+    const std::vector<int> &params=std::vector<int>(), const Format dst_format = JPG) const;
```

Comment:
Here too. Why do we need the same function with swapped parameters?

Reply:
The same reason like the other function. If this is not neccessary i can remove it (swapping the parameters) and only provide cv::imencode params alongside with dst_format.